### PR TITLE
Update ckeditor placeholders in place

### LINKED ISSUE #9640: Switching between answer and comment clears the text field
[https://cea-core.slack.com/archives/C038J512SD8/p1713867427921729](https://cea-core.slack.com/archives/C038J512SD8/p1713867427921729)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/628521446211730/1207132378023169) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by oetherington**
Fixes #9640

We currently have a bug where switching between comment and answer forms deletes the contents of the form.

This is caused by a change made when we implemented the link post UI about a year ago where we had to reinitialise the editor any time the placeholder changed because CKEditor doesn't support updating the placeholder for an editor after it's initialized. However, the most recent relase of CKEditor about two weeks ago fixes this, so the placeholder can now be updated in-situ by an API call.

This PR upgrades to this newest version of CKeditor, as well as making the other required changes to use the same editor for comments and answers and to simply update the editor settings instead of destroying it and creating a new one every time we switch.

This comes with the added advantage of making the new post page a lot snappier when switching between post types.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208364908541086) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/questions/PostsPageQuestionContent.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const PostsPageQuestionContent = ({post, answersTree, refetch}: {
   const author = post.user;
   return (
     <div>
-      {(!currentUser || userIsAllowedToComment(currentUser, post, author, false)) && !post.draft && <NewAnswerCommentQuestionForm post={post} refetch={refetch} />}
+      {(!currentUser || userIsAllowedToComment(currentUser, post, author, false)) && !post.draft && <NewAnswerCommentQuestionForm post={post} />}
```
Comment (oetherington) [inline packages/lesswrong/components/questions/PostsPageQuestionContent.tsx:18]: Unused prop

File: packages/lesswrong/components/editor/EditorFormComponent.tsx [hunk 2]
```diff
@@ -456,7 +447,6 @@ export const EditorFormComponent = ({
       onRestore={onRestoreLocalStorage}
     />}
     <Components.Editor
-      key={placeholderKey.current}
```
Comment (oetherington) [inline packages/lesswrong/components/editor/EditorFormComponent.tsx:459]: This 'key' is what caused the editor to be reinitialized

File: packages/lesswrong/components/editor/CKPostEditor.tsx [hunk 3]
```diff
@@ -431,7 +432,9 @@ const CKPostEditor = ({
   const hasEverDialoguedBefore = !!anyDialogue && anyDialogue.length > 1;
 
   const [_, setEditor] = useContext(EditorContext);
-  
+
+  const [editorObject, setEditorObject] = useState<Editor | null>(null);
```
Comment (oetherington) [inline packages/lesswrong/components/editor/CKPostEditor.tsx:436]: We already have an `editorRef` using `useRef` and the `EditorContext` 2 lines above, but neither seem to do exactly what we want so we add this extra state...

File: packages/lesswrong/components/vulcan-forms/Form.tsx [hunk 4]
```diff
@@ -608,7 +607,18 @@ export class Form<N extends CollectionNameString> extends Component<SmartFormPro
         ) {
           continue;
         }
-        this.setState(getInitialStateFromProps(nextProps));
+        if (prop === "prefilledProps") {
```
Comment (oetherington) [inline packages/lesswrong/components/vulcan-forms/Form.tsx:610]: This change address the `TODO` on line 40 of this file. When changing prefilled props we don't want to reinitialize the editor - just update the relevant fields of the document.

File: packages/lesswrong/components/vulcan-forms/Form.tsx [hunk 5]
```diff
@@ -36,8 +36,7 @@ type FormFieldUnfinished<N extends CollectionNameString> = Partial<FormField<N>>
 // props that should trigger a form reset
 const RESET_PROPS = [
   'collection', 'collectionName', 'typeName', 'document', 'schema', 'currentUser',
-  'fields', 'removeFields',
-  'prefilledProps' // TODO: prefilledProps should be merged instead?
+  'fields', 'removeFields', 'prefilledProps'
```
Comment (s-cheng) [inline packages/lesswrong/components/vulcan-forms/Form.tsx:39]: Perhaps we should update the comment above to note that `prefilledProps` functions differently?

**Review by s-cheng — APPROVED** [review]
Great PR, thanks for doing this! (And thanks for all the helpful comments!)

**Pull request merged**